docs/trace/extending-the-sdk/README.md

```diff
@@ -362,6 +362,14 @@ register custom OpenTelemetry components into their `TracerProvider`s. These
 extension methods can target either the `TracerProviderBuilder` or the
 `IServiceCollection` classes. Both of these patterns are described below.
 
+**Note:** Libraries providing SDK plugins such as exporters, resource detectors,
+and/or samplers should take a dependency on the SDK (`OpenTelemetry.dll`)
+package. Library authors providing instrumentation should take a dependency on
+`OpenTelemetry.Api` or `OpenTelemetry.Api.ProviderBuilderExtensions` package.
+`OpenTelemetry.Api.ProviderBuilderExtensions` exposes interfaces for accessing
+the `IServiceCollection` which is a requirement for supporting the [.NET Options
+pattern](https://learn.microsoft.com/dotnet/core/extensions/options).
```

Comment:
@cijothomas I added this section based on your comments on the other PR. I think it is well enough to keep it simple [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/3a491b14b24c2560e27ff0b74f86359e9e0d4063/src/OpenTelemetry.Api.ProviderBuilderExtensions/README.md?plain=1#L19-L23) and let the link have more details but let me know your thoughts.